Core/src/org/sleuthkit/autopsy/guiutils/WrapLayout.java

```diff
@@ -0,0 +1,607 @@
+                        curX = 0;
+                        // reverse opposite aligned for layout purposes since flipping ltr
+                        Collections.reverse(row.oppositeAligned);
+                        for (Component oppAlignedComp : row.oppositeAligned) {
+                            curX += setComponentDims(oppAlignedComp, useBaseline, !ltr, rowY, rowHeight, leftX, rightX, curX) + getHgap();
+                        }
+                    }
+
+                    rowY += rowHeight + getVgap();
+                }
+            }
+        }
+    }
+
+    /**
+     * Returns the preferred dimensions for this layout given the
+     * <i>visible</i> components in the specified target container.
+     *
+     * @param target The component which needs to be laid out.
+     *
+     * @return The preferred dimensions to lay out the subcomponents of the
+     *         specified container.
+     */
+    @Override
+    public Dimension preferredLayoutSize(Container target) {
+        return layoutSize(target, true);
+    }
+
+    /**
+     * Returns the minimum dimensions needed to layout the <i>visible</i>
+     * components contained in the specified target container.
+     *
+     * @param target The component which needs to be laid out.
+     *
+     * @return The minimum dimensions to lay out the subcomponents of the
+     *         specified container.
+     */
+    @Override
+    public Dimension minimumLayoutSize(Container target) {
+        Dimension minimum = layoutSize(target, false);
+        minimum.width -= (getHgap() + 1);
+        return minimum;
+    }
+
+    /**
+     * This class provides metrics on the parent container dimensions.
+     */
+    private static class ParentDimensions {
+
+        /**
+         * The full width that the component can consume.
+         */
+        final int outerWidth;
+
+        /**
+         * The full width that subcomponent rows can consume. This is the
+         * outerWidth accounting for left and right insets.
+         */
+        final int innerWidth;
+
+        /**
+         * The insets of the parent container.
+         */
+        final Insets insets;
+
+        ParentDimensions(int outerWidth, int innerWidth, Insets insets) {
+            this.outerWidth = outerWidth;
+            this.innerWidth = innerWidth;
+            this.insets = insets;
+        }
+    }
+
+    /**
+     * Derives metrics on the space allowed within the parent container for rows
+     * of components.
+     *
+     * @param target The target container.
+     *
+     * @return The dimensions for laying out components.
+     */
+    private ParentDimensions getTargetDimensions(Container target) {
+        //  Each row must fit with the width allocated to the containter.
+        //  When the container width = 0, the preferred width of the container
+        //  has not yet been calculated so lets ask for the maximum.
+
+        int targetWidth = target.getSize().width;
```

Comment:
Assigned value never used.

Reply:
To which variable are you referring?  I believe targetWidth is used a couple times in getTargetDimensions.